### PR TITLE
Ruby Project: Solo

### PR DESCRIPTION
**Pull request by rkyankya**
Enumerables

### CODE REVIEW

**Comment by Ispirett** [comment]
**Please use Gitflow for this project and resubmit for review.**

File: enum_methods.rb [hunk 1]
```diff
@@ -0,0 +1,273 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
```
Comment (mcrd25) [inline enum_methods.rb:4]: `my_each` should return enumerator when block not given.

File: enum_methods.rb [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    if block_given?
+      for i in 0..self.length - 1
+        yield(self[i])
+      end
+    end
+  end
+
+  def my_each_with_index
```
Comment (mcrd25) [inline enum_methods.rb:12]: `my_each_with_index` should return enumerator when block not given.

File: enum_methods.rb [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    if block_given?
+      for i in 0..self.length - 1
+        yield(self[i], i)
+      end
+    end
+  end
+
+  def my_select
```
Comment (mcrd25) [inline enum_methods.rb:20]: `my_select` should return enumerator when block not given.

File: enum_methods.rb (hunk 2 above)
Comment (mcrd25) [reply enum_methods.rb:12]: Additional Requirements for this method:
it should:
- return an Enumerator if no block is given

File: enum_methods.rb (hunk 1 above)
Comment (mcrd25) [reply enum_methods.rb:4]: Additional Requirements for this method:
it should:
- return an Enumerator if no block is given

File: enum_methods.rb (hunk 3 above)
Comment (mcrd25) [reply enum_methods.rb:20]: Additional Requirements for this method:
it should:
- return an Enumerator if no block is given

File: enum_methods.rb [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+      for i in 0..self.length - 1
+        if yield(self[i])
+          r.push(self[i])
+        end
+      end
+      r
+    end
+  end
+
+  def my_all?(arg1 = nil)
```
Comment (mcrd25) [inline enum_methods.rb:32]: Additional Requirements for this method:
it should:
- return true when no block or argument is given and none of the collection members are false or nil
- return true when a class is passed as an argument and all of the collection is a member of such class
- return true when a Regex is passed as an argument and all of the collection matches the Regex
- return true when a pattern other than Regex or a Class is given and if all of the collection matches the pattern

Comment (mcrd25) [reply enum_methods.rb:32]: [OPTIONAL]
Also, this is a really long method!
Surely there is way to shorten it? 
You can use your my_each and make simpler comparisons.

File: enum_methods.rb [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+        if ln == cnt
+          true
+        else
+          false
+        end
+      end
+    end
+  end
+
+  def my_any?
```
Comment (mcrd25) [inline enum_methods.rb:78]: Additional Requirements for this method:
it should:
- return true when no block or argument is given and at least one of the collection members are false or nil
- return true when a class is passed as an argument and at least one of the collection is a member of such class
- return true when a Regex is passed as an argument and at least one of the collection matches the Regex
- return true when a pattern other than Regex or a Class is given and if at least one of the collection matches the pattern

File: enum_methods.rb [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+      for i in 0..self.length - 1
+        if yield(self[i])
+          return true
+        end
+      end
+    end
+    false
+  end
+
+  def my_none?
```
Comment (mcrd25) [inline enum_methods.rb:89]: Additional Requirements for this method:
it should:
- return true when no block or argument is given and none of the collection members are true
- return true when a class is passed as an argument and not one of the collection is a member of such class
- return true when a Regex is passed as an argument and not one of the collection matches the Regex
- return true when a pattern other than Regex or a Class is given and if not one of the collection matches the pattern

File: enum_methods.rb [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+        true
+      else
+        false
+      end
+    else
+      nil
+    end
+  end
+
+  def my_count(obj = nil)
```
Comment (mcrd25) [inline enum_methods.rb:109]: [OPTIONAL]
using my_each may decrease the length of this method.

File: enum_methods.rb [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+          if obj == self[i]
+            cnt += 1
+          end
+        end
+        return cnt
+      end
+    end
+  end
+
+  def my_map
```
Comment (mcrd25) [inline enum_methods.rb:134]: Additional Requirements for this method:
it should:
- return an Enumerator if no block is given

File: enum_methods.rb [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+    if block_given?
+      for i in 0..self.length - 1
+        self[i] = yield(self[i])
+      end
+      self
+    end
+  end
+
+  def my_inject(arg1 = 1, sbl = nil)
```
Comment (mcrd25) [inline enum_methods.rb:143]: Your inject does not pass any of our tests 😢 

It should:
- return combination of all elements of enum by applying a binary operation, specified by a block
- return combination of each element of the collection by applying the symbol as a named method when a symbol is specified

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required

Hi @rkyankya 👋,

Please close this pull request and resubmit using [Git Flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/).
***How Git Flow Works***
  - You'd have the `master` branch purely for production purposes 
  - the `development` branch would act as the main working branch where you create merge requests from `feature` branches to `development` for reviews on these requests
  - If you are requesting a review for a project without multiple milestones simply have one main `feature` branch (appropriately named) that you can then branch off from and merge to accordingly as you complete your project. 
  - After you have completed your project and it's ready for your first review you can then request a code review form your main `feature` branch to `development` branch
  - Once the pull request is approved you can then merge this pull request
  - You would then merge the `development` branch to `master` branch 

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Pull request closed without merging**